Split user keys for db access

nklaassen opened the pull request:
This PR is part of the implementation of [RFD 136](https://github.com/gravitational/teleport/blob/master/rfd/0136-modern-signature-algorithms.md).

The main change here is that `tsh` now uses a unique private key every time it gets a new db cert issued. This new key will use a signature algorithm according to the cluster's currently configured `signature_algorithm_suite`.

Changelog: Changed the certificate and private key file paths for application, database, and Kubernetes access.

nklaassen committed:
split user keys for db access

flyinghermit reviewed (approved):
I only reviewed code level changes and those changes LGTM.

Please also get a review from database access team.

greedy52 reviewed (approved):
Overall LGTM. Thanks!

Several comments:
1. Looks like `lib/teleterm` does not need an update? (as recently we changed it to use TLS cert in memory instead of paths https://github.com/gravitational/teleport/blob/39b2d0f5279b053527db092d4cb4c128b74f99a2/lib/teleterm/clusters/cluster_databases.go#L135)
1. The team will test out all databases during release testing.
1. Since some paths changed, this would be a breaking change for existing scripts that assumes the key/cert path. Might worth a small note on the release change log?

nklaassen commented on tool/tsh/common/db_test.go:
```diff
@@ -441,7 +441,7 @@ func testDatabaseLogin(t *testing.T) {
 			args := append([]string{
 				// default --db-user and --db-name are selected from roles.
-				"db", "login",
+				"db", "login", "--insecure",
```
I introduced a Ping that hits the proxy and requires `--insecure` flag for this test, `mustLogin` already used for the initial login uses `--insecure` as well

nklaassen commented:
> Overall LGTM. Thanks!
> 
> Several comments:
> 
> 1. Looks like `lib/teleterm` does not need an update? (as recently we changed it to use TLS cert in memory instead of paths https://github.com/gravitational/teleport/blob/39b2d0f5279b053527db092d4cb4c128b74f99a2/lib/teleterm/clusters/cluster_databases.go#L135
>    )

Yep, looks like it's fine. 

> 2. The team will test out all databases during release testing.

Thanks! I'm counting on it

> 3. Since some paths changed, this would be a breaking change for existing scripts that assumes the key/cert path. Might worth a small note on the release change log?

I'm planning to add a changelog entry for the whole "configurable algorithm suites" feature later, but these seems like a good point to mention the changed file paths, added a CL entry to the PR description

nklaassen commented on lib/client/profile.go:
```diff
@@ -444,7 +444,7 @@ func (p *ProfileStatus) KeyPath() string {
 // certificate for this profile, for the specified cluster.
 //
-// It's kept in <profile-dir>/keys/<proxy>/<user>-db/<cluster>/<name>-x509.pem
+// It's kept in <profile-dir>/keys/<proxy>/<user>-db/<cluster>/<name>.crt
```
Older clients will still read and write to that path, new clients will read/write the new path, and since they don't overlap at all, after an upgrade/downgrade it will just look like you aren't logged into any DBs and a fresh login should be triggered. We're not really doing anything for back compat here so I don't think we need to mention the path that old versions will use
